rocAL/rocAL/source/coco_file_source_reader.cpp

```diff
@@ -56,38 +63,94 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
     //    std::cout<<"Metadata reader not initialized for COCO file source\n";
 
     ret = subfolder_reading();
+
+    // if _keypoint is set we need to process by annotation. Iterate the map and for every annotation
+    // check if the image is present in the path user shared. If so add it in a new temp list and assign
+    // it to the _file_names list at the end.
+    if(_keypoint)
+    {
+        std::vector<std::string> temp_file_names;
+        for (auto &elem : _annotation_image_key_map)
+        {
+            auto itr = std::find(_files.begin(), _files.end(), elem.second);
+            if(itr != _files.end())
+            {
+                int index = std::distance(_files.begin(), itr);
+                _annotation_ids.push_back(elem.first);
```

Comment:
reserve annotation ids and temp_file_names so that pushback doesn't allocate every time.

Reply:
Done